core/src/main/java/com/whylogs/core/ColumnProfile.java

```diff
@@ -31,28 +33,52 @@
 public class ColumnProfile {
   public static final int FREQUENT_MAX_LG_K = 7;
   private static final int CARDINALITY_LG_K = 12;
+  private static Map<String, String> ENV;
+  private static ImmutableList<String> nullEnvStrs;
 
   @NonNull private final String columnName;
   @NonNull private final CountersTracker counters;
   @NonNull private final SchemaTracker schemaTracker;
   @NonNull private final NumberTracker numberTracker;
   @NonNull private final ItemsSketch<String> frequentItems;
   @NonNull private final HllSketch cardinalityTracker;
+  private final ImmutableList<String> nullStrs;
+
+  static synchronized ImmutableList<String> nullStrsFromEnv() {
+    if (ColumnProfile.ENV == null) {
+      ColumnProfile.ENV = System.getenv();
+      String nullSpec = ColumnProfile.ENV.get("NULL_STRINGS");
+      ColumnProfile.nullEnvStrs =
+          nullSpec == null ? null : ImmutableList.copyOf(nullSpec.split(","));
+    }
+    return ColumnProfile.nullEnvStrs;
+  }
 
   public ColumnProfile(String columnName) {
+    this(columnName, nullStrsFromEnv());
+  }
+
+  public ColumnProfile(String columnName, ImmutableList<String> nullStrs) {
```

Comment:
```suggestion
  public ColumnProfile(String columnName, @NonNull ImmutableList<String> nullStrs) {
```

Reply:
ok but then we will be converting every value to string and testing against an empty set in the critical code path.

Reply:
Empty set `contains` should be very fast :). I wouldn't worry about that. While performance is important, I would prioritize clean code + ease of understanding + testing over little performance gain

```
    private static class EmptySet<E>
        extends AbstractSet<E>
        implements Serializable
    {
        public boolean contains(Object obj) {return false;}
```

Reply:
> ok but then we will be converting every value to string 

Can you clarify this comment?